compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUTensorCoreVectorization.cpp

```diff
@@ -96,33 +94,36 @@ struct LLVMGPUTensorCoreVectorizationPass
 
       // Step 3. Prepare vector operations to be lowered to native tensor core
       // operations (nvgpu.mmasync, nvgpu.ldmatrix).
-      if (llvmgpuUseMMASync) {
+      if (tensorCoreType == GPUTensorCoreType::MMA_SYNC) {
         RewritePatternSet vectorContractPatterns(funcOp.getContext());
         mlir::vector::populateCastAwayVectorLeadingOneDimPatterns(
             vectorContractPatterns);
         mlir::populatePrepareVectorToMMAPatterns(vectorContractPatterns,
-                                                 llvmgpuUseMMASync);
+                                                 /*useMMASync=*/true);
         if (failed(applyPatternsAndFoldGreedily(
                 getOperation(), std::move(vectorContractPatterns)))) {
           return signalPassFailure();
         }
       }
-
+      bool useMmaSyncShape = tensorCoreType == GPUTensorCoreType::MMA_SYNC;
```

Comment:
`useMmaSyncShape`? Trying to understanding the meaning of shape here, I think it is just a boolean to make sure if Tensor Cores are using `wmma` path or `mma.sync` path. Would `isMmaSyncEnabled` read better?

Reply:
Shape in the sense that it is used to unroll to a native shape. The reason why there is a difference in unrolling is that the native shape is different between wmma and mma.sync. Does that make sense?

Reply:
Yeah, the shapes are different, but the other differences in handling Native Tensor Core support vs. through WMMA API are the following:
- Programming Tensor Core natively using `mma.sync` vs. `wmma`
- Loading using `ldmatrix` vs. `wmma.load`
- Unrolling and considering the layout till thread-level vs. stop the unrolling till `wmma` API calls. No need to unroll till the thread-level and apply thread-layout mapping, the WMMA API will do it for us, it won't be the most efficient.
- Shapes are different for `mma.sync` vs. `wmma` for different data types.

Reading`useNativeTensorCores` or `isMmaSyncEnabled` goes better for me just the [NVIDIA terminology](https://github.com/NVIDIA/cutlass/releases/tag/v2.0.0). It is ok to choose whatever naming works better here for current codebase.

Reply:
but the unrolling doesn't care about that right? Here this flag is just to control unrolling, so the only thing we care about is getting the right shapes for mma.sync op?

Reply:
aah ok. This function only control the shapes that are applied during the unrolling during Tensor Core vectorization.